src/core/border_router/routing_manager.cpp

```diff
@@ -568,6 +628,34 @@ void RoutingManager::DeprecateOnLinkPrefix(void)
                                         TimeMilli::SecToMsec(kDefaultOnLinkPrefixLifetime));
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
+void RoutingManager::EvaluateNat64Prefix(void)
+{
+    OT_ASSERT(mIsRunning);
+
+    NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
+    NetworkData::ExternalRouteConfig config;
+    bool                             hasAdvertisedNat64Prefix = false;
+
+    otLogInfoBr("Evaluating NAT64 prefix");
+
+    while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, config) == kErrorNone)
+    {
+        if (config.mNat64 && config.GetPrefix().IsValidNat64())
+        {
+            hasAdvertisedNat64Prefix = true;
+        }
+    }
+
+    if (!hasAdvertisedNat64Prefix)
```

Comment:
```suggestion
    if (!hasAdvertisedNat64Prefix && !mIsAdvertisingLocalNat64Prefix)
```
nit: unnecessary to add external route if it's already locally advertised.

Reply:
Updated.